src/lib/operators/jit_operator/operators/jit_compute.cpp

```diff
@@ -0,0 +1,213 @@
+      jit_compute(jit_less_than, left_value, right_value, result_value);
+      break;
+    case ExpressionType::LessThanEquals:
+      jit_compute(jit_less_than_equals, left_value, right_value, result_value);
+      break;
+    case ExpressionType::Like:
+      jit_compute(jit_like, left_value, right_value, result_value);
+      break;
+    case ExpressionType::NotLike:
+      jit_compute(jit_not_like, left_value, right_value, result_value);
+      break;
+
+    case ExpressionType::And:
+      jit_and(left_value, right_value, result_value);
+      break;
+    case ExpressionType::Or:
+      jit_or(left_value, right_value, result_value);
+      break;
+
+    default:
+      Fail("expression type not supported");
+  }
+}
+
+std::pair<const DataType, const bool> JitExpression::_compute_result_type() {
+  const auto left_result = _left_child->result();
+
+  if (!_is_binary_operator()) {
+    switch (_expression_type) {
+      case ExpressionType::Not:
+        return std::make_pair(DataType::Bool, left_result.is_nullable());
+      case ExpressionType::IsNull:
+      case ExpressionType::IsNotNull:
+        return std::make_pair(DataType::Bool, false);
+      default:
+        Fail("expression type not supported");
+    }
+  }
+
+  const auto right_result = _right_child->result();
+  DataType result_data_type;
+
+  switch (_expression_type) {
+    case ExpressionType::Addition:
+      result_data_type = jit_compute_type(jit_addition, left_result.data_type(), right_result.data_type());
+      break;
+    case ExpressionType::Subtraction:
+      result_data_type = jit_compute_type(jit_subtraction, left_result.data_type(), right_result.data_type());
+      break;
+    case ExpressionType::Multiplication:
+      result_data_type = jit_compute_type(jit_multiplication, left_result.data_type(), right_result.data_type());
+      break;
+    case ExpressionType::Division:
+      result_data_type = jit_compute_type(jit_division, left_result.data_type(), right_result.data_type());
+      break;
+    case ExpressionType::Modulo:
+      result_data_type = jit_compute_type(jit_modulo, left_result.data_type(), right_result.data_type());
+      break;
+    case ExpressionType::Power:
+      result_data_type = jit_compute_type(jit_power, left_result.data_type(), right_result.data_type());
+      break;
+    case ExpressionType::Equals:
+    case ExpressionType::NotEquals:
+    case ExpressionType::GreaterThan:
+    case ExpressionType::GreaterThanEquals:
+    case ExpressionType::LessThan:
+    case ExpressionType::LessThanEquals:
+    case ExpressionType::Like:
+    case ExpressionType::NotLike:
+    case ExpressionType::And:
+    case ExpressionType::Or:
+      result_data_type = DataType::Bool;
+      break;
+    default:
+      Fail("expression type not supported");
+  }
+
+  return std::make_pair(result_data_type, left_result.is_nullable() || right_result.is_nullable());
+}
+
+bool JitExpression::_is_binary_operator() const {
```

Comment:
I am not a fan of free-floating functions, but maybe we can change `AbstractExpression<DerivedExpression>::is_binary_operator()` to `is_binary_operator(ExpressionType)` and save us the code duplication?